Add follow up Neurohackademy blog post (#292)
2i2c-org/2i2c-brochure: content/blog/2024/neurohackademy-summer-school-2/index.md

---
title: Neurohackademy Summer School Reflections
date: "2024-06-21"
authors: ["Ariel Rokem", "Jenny Wong"]
tags: [bioscience, education]
categories: [impact]
featured: false
draft: false
---

_Thank you to Ariel Rokem for guest writing this blog post!_

![Landing page of the Neurohackademy Summer School website](featured.png "[Neurohackademy Summer School](https://neurohackademy.org/)")

## Subsection

> Text goes here.

Reviewer: Some initial bullet points:

```suggestion

- What is NeuroHackademy? 
   -  A word or two about the hybrid structure (see also https://pubmed.ncbi.nlm.nih.gov/38763989/). 
- The role of 2i2c and the Jupyterhub in the NeuroHackademy curriculum. 
- Some features that we've honed over the years: 
    - nbgitpuller from an organization curriculum repo, with a ragtag team of instructors as collaborators.
    - Shared data storage with user read and admin read-write.
    - Emphasis on cloud-hosted datasets (and maybe a word or two about how much easier and abundant that has become!).
    - 
- This year's unique features: 
    - "Bring your own image" together with repo2docker and Github actions as a way to prototype changes to the image. 
    - Pull requests on 2i2c infrastructure. Organizers as GitHub collaborators. 
    - GPU instances. 
    
- Some lessons learned and desiderata for next year: 
    - jupyter keepalive for long-running processes.
    - ??? 
```

---
title: NeuroHackademy Summer School Reflections
date: "2024-08-19"
authors: ["Ariel Rokem", "Jenny Wong"]
tags: [bioscience, education]
categories: [impact]
featured: false
draft: false
---

_Thank you to Ariel Rokem for guest writing this blog post!_

![Landing page of the NeuroHackademy Summer School website](featured.png "[NeuroHackademy Summer School](https://NeuroHackademy.org/)")

## What is NeuroHackademy?

> NeuroHackademy is a summer school hosted by the [University of Washington eScience Institute](http://escience.washington.edu/). The school is aimed at providing students with skills to work with open source tools and workflows for analyzing human neuroscience data in an effort to make scientific analysis and results shareable, reproducible, and accessible.

Reviewer: ```suggestion
Part summer school, part free-wheeling hackathon, all focused on the use of data science methods in neuroscience, NeuroHackademy is an event that was recently hosted by the [University of Washington eScience Institute](http://escience.washington.edu/) in Seattle, WA, USA. This event, that has been running annually since 2016, aims to provide early-career researchers in Psychology, Medicine, Neuroscience, and other related fields with the skills and knowledge that they need to effectively and rigorously work with open source tools and workflows for analyzing human neuroscience data. This supports the effort to make scientific analysis and results shareable, reproducible, and accessible.
```

## Global and inclusive

> The hybrid format of the summer school that allows the participation of students from all backgrounds. This aspect plays an important part in building a global and inclusive community of practice. See the paper [Hands-On Neuroinformatics Education at the Crossroads of Online and In-Person: Lessons Learned from NeuroHackademy](https://pubmed.ncbi.nlm.nih.gov/38763989/) to read more on this subject.

Reviewer: ```suggestion
In 2020, the event had to rapidly pivot to an online format, and this format was also used in 2021. Through this experience, the organizers (Ariel Rokem and Noah Benson) realized that many participants preferred the online format. For example, participants who could not afford to travel to Seattle, or participants who had care-taking responsibilities that precluded them from participating in a two-week event away from their homes.  In 2022, the event pioneered a hybrid format, where half of the participants are present in-person and half join the event via zoom, slack, GitHub, and of course through a dedicated 2i2c JupyterHub. Taken together, this format allows the participation of students from a larger range of backgrounds and locations. This aspect plays an important part in building a global and inclusive community of practice. See the paper [Hands-On Neuroinformatics Education at the Crossroads of Online and In-Person: Lessons Learned from NeuroHackademy](https://pubmed.ncbi.nlm.nih.gov/38763989/) to read more on this subject.
```

## Collaboration with 2i2c

### Previous years

> 2i2c has operated a JupyterHub for the last X years that provides an interactive computing platform for learners (see [blog post](blog/2024/NeuroHackademy-summer-school/index.md) announcing support for this year's event).

Author: ```
2i2c has operated a JupyterHub for the last X years that provides an interactive computing platform for learners (see [blog post](blog/2024/NeuroHackademy-summer-school/index.md) announcing support for this year's event).
```

@arokem Can you fill the gap for "... X years ..." please?

> 2i2c has operated a JupyterHub for the last X years that provides an interactive computing platform for learners (see [blog post](blog/2024/NeuroHackademy-summer-school/index.md) announcing support for this year's event).

Reviewer: ```suggestion
NeuroHackademy has been an early adopter of the cloud-based JupyterHub model, setting up its first hub using the zero-to-jupyterhub guide in 2018. NeuroHackademy partnered with 2i2c as soon as it was founded, and 2i2c has operated a JupyterHub for the last 3 years. The hub provides an interactive computing platform for learners, and implements the "digital watering hole" for practical and immediate access to a range of cloud-based datasets in human neuroscience (see [blog post](blog/2024/NeuroHackademy-summer-school/index.md) announcing support for this year's event).
```

In terms of the software environment, the following tools and features that have benefited the event over the years include

- [`nbgitpuller`](https://github.com/jupyterhub/nbgitpuller) allows students to synchronise lesson content with an organizational GitHub repository that is collaboratively maintained by the lesson instructors.
- [Shared data file storage](https://docs.2i2c.org/user/topics/data/sharing/) with read-only access for learners and read-write access for instructors
- Access to an abundance of neuroimaging data hosted in cloud object storage
  - [The Human Connectome Project](https://www.humanconnectome.org/)
  - [The Natural Scenes Dataset](https://naturalscenesdataset.org/)
  - [OpenNeuro](https://openneuro.org/)

> 

Reviewer: ```suggestion
    - [The Healthy Brain Network](https://fcp-indi.s3.amazonaws.com/index.html#data/Projects/HBN/)
    - And more.
```

### This year

This year 2i2c supported the following tools and features for NeuroHackademy

> - A "Bring your own image" option where users can pull any image hosted on a container registry into the hub. See our [Integrating BinderHub with JupyterHub: Empowering users to manage their own environments](blog/2024/jupyterhub-binderhub-gesis) blog post for more details

Reviewer: ```suggestion
- A "Bring your own image" option where users can pull any image hosted on a container registry into the hub. See our [Integrating BinderHub with JupyterHub: Empowering users to manage their own environments](blog/2024/jupyterhub-binderhub-gesis) blog post for more details.
```

> - [`repo2docker`](https://github.com/jupyterhub/repo2docker) and GitHub actions to build and prototype images from a repository

Reviewer: ```suggestion
- [`repo2docker`](https://github.com/jupyterhub/repo2docker) and GitHub actions to build and prototype images from a repository.
```

> - The support services provided by 2i2c and the ability for instructors to [open pull requests on 2i2c infrastructure](https://infrastructure.2i2c.org/contributing/community-partner/) for speedy resolution

Reviewer: ```suggestion
- The support services provided by 2i2c and the ability for instructors to [open pull requests on 2i2c infrastructure](https://infrastructure.2i2c.org/contributing/community-partner/) for speedy resolution.
```
